src/classes/GE_PaymentServices.cls

```diff
@@ -122,13 +122,23 @@ public with sharing class GE_PaymentServices {
         return config.get(PS_IntegrationServiceConfig.BASE_URL);
     }
 
-    public String getJwt() {
-        JWTPayload jwtPayload = new JWTPayload(config.get(PS_IntegrationServiceConfig.PRODUCTID));
+    public String getJwtForPurchase() {
+        JWTPayload jwtPayload = new JWTPayload(config.get(PS_IntegrationServiceConfig.PRODUCTID),
+                PS_IntegrationServiceConfig.PURCHASE_ROLE);
         String jwt = UTIL_Jwt.getSignedJWTFromString(
                 jwtPayload.getAsString(),
                 config.get(PS_IntegrationServiceConfig.API_KEY)
         );
+        return jwt;
+    }
 
+    public String getJwtForToken() {
```

Comment:
quick explanation of why `getJwtForToken()` and `getJwtForPurchase()` are being used with some duplicate code. We have to pass an explicit role to get a JWT now since we are generating jwts for more than one role. Since we're using the SDK to generate the JWT for tokenize we can't pass the role as a parameter since that would require Visualforce remoting, which would require the method to be `static` which it isn't and we are using a constructor Builder pattern moving forward for the PS_Request class anyway. That's going to make tokenizing from the SDK a special case. We likely can make the regular `getJwt()` function more generic once we merge with @RadSFDeveloper's code but I have created it this way for now for simplicity sake until we merge with her code.

Reply:
Everything looks good @dfuller114 . Wondering if additional change to [PS_TestPaymentServices](https://github.com/SalesforceFoundation/NPSP/blob/0bd334447467506e0944c62efe5e98c1e9355b04/unpackaged/config/ps_testing/classes/PS_TestPaymentService.cls#L66) is required to handle the CAPTCHA requirements?